src/WorldStorage/NBTChunkSerializer.cpp

```diff
@@ -541,22 +541,22 @@ void cNBTChunkSerializer::AddMonsterEntity(cMonster * a_Monster)
 			case mtEnderman:
 			{
 				const cEnderman *Enderman = reinterpret_cast<const cEnderman *>(a_Monster);
-				m_Writer.AddShort("carried",     (Int16) Enderman->GetCarriedBlock());
-				m_Writer.AddShort("carriedData", (Int16) Enderman->GetCarriedMeta());
+				m_Writer.AddShort("carried",     static_cast<Int16>(Enderman->GetCarriedBlock()));
+				m_Writer.AddShort("carriedData", static_cast<Int16>(Enderman->GetCarriedMeta()));
 				break;
 			}
 			case mtHorse:
 			{
 				const cHorse *Horse = reinterpret_cast<const cHorse *>(a_Monster);
-				m_Writer.AddByte("ChestedHorse",   Horse->IsChested());
-				m_Writer.AddByte("EatingHaystack", Horse->IsEating());
-				m_Writer.AddByte("Tame",           Horse->IsTame());
+				m_Writer.AddByte("ChestedHorse",   (Horse->IsChested())? 1 : 0);
+				m_Writer.AddByte("EatingHaystack", (Horse->IsEating())? 1 : 0);
+				m_Writer.AddByte("Tame",           (Horse->IsTame()));
```

Comment:
No need for the extra parentheses. Compare:
`(a == b) ? 1 : 0`
`boolVal ? 1 : 0`